article/article.Rmd

```diff
@@ -0,0 +1,322 @@
+(i.e. the day in which a battle started or ended is unknown)
+or come from unreliable sources (i.e. the date might not be trustworthy).
+
+```{r data, warning=FALSE, message=FALSE}
+battles <- messydates::battles
+battles
+```
+
+\pkg{messydates} facilitates working with these dates as we can, for example,
+check which dates are precise, get the median values for imprecise dates,
+and find the longest battle in 2001.
+
+```{r working with dates, warning=FALSE, message=FALSE}
+messydates::is_precise(battles$Date)
+as.Date(battles$Date, median)
+as.numeric(as.Date(battles$Date, max) - as.Date(battles$Date, min))
+```
+
+Getting the timing can be important for researchers, however,
+when faced with date imprecision, researchers usually have to chose
+between making arbitrary choices (e.g. adding "-01-01" to all incomplete dates)
+or work imprecise dates (e.g. keep the year only).
+Yet, both choices may lead to bias results.
+This is especially true if researchers' are looking to generate inferences.
+Let's assume, for instance, certain researcher is interested in the
+relationship between the United States (US) being a party to a conflict and the
+duration of the conflict in 2001.
+The researcher hypothesizes that conflicts involving the US have
+a shorter duration because the US has the most powerful military in the world.
+A relationship could be mediated by the number of parties involved in the conflict.
+Using \pkg{messydates}, we can create two different date variables
+in the battles data to be our dependent variables representing conflict time,
+one variable with an arbitrary cut of point and the other variable
+with random values for uncertain or approximate dates.
+The ability to resolve uncertain or approximate dates differently is
+particularly useful for checking the robustness of results.
+As our independent variable, we create a dummy variable for whether the US
+was involved in the conflict.
+As our control, we code the number of actors in the conflict.
+With these variables we build two linear regression models,
+one using the arbitrary dates and the other the random dates.
+
+```{r lm, warning=FALSE, message=FALSE}
+set.seed(1301)
+battles <- battles %>%
+  mutate(arbitrary = as.numeric(as.Date(Date, max) - as.Date(Date, min)),
+         random = ifelse(grepl("\\?|\\~", Date),
+                         abs(as.Date(Date, random) - as.Date(Date, random)),
+                         arbitrary),
+         US_party = ifelse(grepl("US", Parties), 1, 0),
+         n_actors = c(2, 2, 2, 2, 2, 3, 2, 4, 4, 4, 3, 3, 4, 4, 5, 4, 3, 3, 2, 2))
+lm(arbitrary ~ US_party + n_actors, battles)
+lm(random ~ US_party + n_actors, battles)
+```
+
+Notice how the regression coefficients change when we pick random values
+within the range for the uncertain and approximate dates in the battles data,
+in comparison to setting arbitrary cut off points.
+Although not statistically significant, the coefficient for US being a party
```

Comment:
So is the fluctuation statistically significant?

Reply:
Good point, but this is hard to say since we are changing the DV in this case... Perhaps I can rework the problem/hypothesis formulation so that the IVs are the ones changing instead and then we can generate such information.

Reply:
Could test whether there is a significant outlier to this collection of 'observations'

Reply:
I added a bit about outliers for now... I am not sure this will do or whether we will need to change the formulation of the case study a bit. However, I do like the battles dataset and think we should try to use it in any case.